app.py

```diff
@@ -186,16 +195,32 @@ def editpost(id):
             post.body = form.body.data
             post.uploaded = datetime.now()
             db.session.commit()
-            return redirect(url_for('single_post', id = id))
-    return render_template('editpost.html', form = form, post = post)
+            return redirect(url_for('single_post', id=id))
+    return render_template('editpost.html', form=form, post=post)
 
 
-@app.route('/singlepost/<id>', methods = ['POST', 'GET'])
-@login_required
+@app.route('/singlepost/<id>', methods=['GET'])
 def single_post(id):
-    post = Posts.query.filter_by(id = id).first()
-    comments = post.comments.all()
-    return render_template('single_post.html', post = post, comments = comments)
+    post = Posts.query.filter_by(id=id).first()
+    if post:
+        comments = post.comments.all()
+        post.views += 1
```

Comment:
this is technically just a tiny bit unsafe, and I learned thist oo. what we should be doing is something like `post.views = Post.views + 1`

https://stackoverflow.com/questions/2334824/how-to-increase-a-counter-in-sqlalchemy#comment30697297_2334917